.github/workflows/update-platform-branch.yaml

```diff
@@ -0,0 +1,104 @@
+#
+# To test:
+#   `act workflow_dispatch -W ./.github/workflows/update-platform-branch.yaml --input tag=protocol/go/v0.3.1`
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "The new tag or branch to update the platform.branch property to use for targeting the RPC protocol buffers."
+        required: true
+        default: "protocol/go/v0.3.0"
+
+jobs:
+  update-platform-branch:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+      actions: read
+
+    steps:
+      - name: Checkout java-sdk repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: true # Required for "commit changes" and push steps below
+
+      - name: Validate tag as a valid git ref
+        run: |
+          if ! echo "$TAG" | grep -Eq "^[a-zA-Z0-9._\-\/]+$"; then
+            echo "Invalid tag format: [$TAG]"
+            exit 1
+          fi
+          echo "TAG=$TAG" >> $GITHUB_ENV
+        env:
+          TAG: ${{ github.event.inputs.tag }}
+
+      - name: Check if tag exists in the repository
+        run: |
+          if ! git ls-remote --exit-code --heads --tags https://github.com/opentdf/platform.git "$TAG"; then
+            echo "Tag or branch [$TAG] does not exist in the platform repository."
+            exit 1
+          fi
+
+      - name: Update platform.branch in pom.xml files
+        run: |
+            find . -name "pom.xml" -exec sed -i.bak "s|<platform.branch>.*</platform.branch>|<platform.branch>${TAG}</platform.branch>|g" {} \;
+            CHANGED_FILES=$(find . -name "pom.xml" -exec diff -u {} {}.bak \;)
+            if [ -z "$CHANGED_FILES" ]; then
+              echo "No changes detected in pom.xml files." | tee -a $GITHUB_STEP_SUMMARY
+              find . -name "pom.xml.bak" -delete
+              exit 1
+            fi
+            echo "The following pom.xml files were updated: $CHANGED_FILES"
+            find . -name "pom.xml.bak" -delete
+
+      - name: Fetch release notes for the tag
+        id: fetch-release-notes
+        run: |
+          RELEASE_URL="https://api.github.com/repos/opentdf/platform/releases/tags/$TAG"
+          RESPONSE=$(curl -s -H "Accept: application/vnd.github.v3+json" "$RELEASE_URL")
+          RELEASE_NOTES=$(echo "$RESPONSE" | jq -r '.body // ""')
+          if [ "$RELEASE_NOTES" = "" ]; then
+            echo "No release notes found for tag $TAG."
+          fi
+          echo "RELEASE_NOTES=$RELEASE_NOTES" >> $GITHUB_ENV
+          echo "RELEASE_URL=https://github.com/opentdf/platform/releases/tag/$TAG" >> $GITHUB_ENV
+
+      - name: Commit changes
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git checkout -b update-platform-branch
```

Comment:
should the created branch be unique in case the PR hasnt been merged? i would assume if there were already changes on update-platform-branch you may run into merge conflict or something when commiting?

Reply:
It probably doesn't make sense to have more than one of these active at a time. I'm pretty sure the checkout here will fail if it exists? So we should check for an existing PR and update that instead, yeah

Reply:
Better?